src/superannotate/lib/core/usecases.py

```diff
@@ -3249,56 +3249,57 @@ def execute(self):
             if not image_data:
                 raise AppException("There is no images to attach annotation.")
             image_data = image_data[0]
-            auth_data = self._backend_service.get_annotation_upload_data(
+            response = self._backend_service.get_annotation_upload_data(
                 project_id=self._project.uuid,
                 team_id=self._project.team_id,
                 folder_id=self._folder.uuid,
                 image_ids=[image_data["id"]],
             )
-            session = boto3.Session(
-                aws_access_key_id=auth_data["creds"]["accessKeyId"],
-                aws_secret_access_key=auth_data["creds"]["secretAccessKey"],
-                aws_session_token=auth_data["creds"]["sessionToken"],
-                region_name=auth_data["creds"]["region"],
-            )
-            resource = session.resource("s3")
-            bucket = resource.Bucket(auth_data["creds"]["bucket"])
-            self.fill_classes_data(self._annotations)
-            bucket.put_object(
-                Key=auth_data["images"][str(image_data["id"])]["annotation_json_path"],
-                Body=json.dumps(self._annotations),
-            )
-            if (
-                self._project.project_type == constances.ProjectType.PIXEL.value
-                and os.path.exists(self._annotation_path)
-            ):
-                with open(self._annotation_path, "rb") as descriptor:
-                    file = io.BytesIO(descriptor.read())
+            if response.ok:
```

Comment:
If not response.ok:
   return

Reply:
This is something like a rule of use case def
`def execute(): `
`    if self._is_valid():`
          code_block
          response.data  = ...
      return response
`